from protocol.errors import EOPINotImplementedError


class Problem4P1(object):
    """
    Note: we can potentially change this function to accept an array of numbers
    in case we find out that argument "num" can't be big enough. But apparently
    in python there's no limit to how big a number can be! source:
    https://www.geeksforgeeks.org/what-is-maximum-possible-value-of-an-integer-in-python/
    """
    def get_parity(self, num):
        raise EOPINotImplementedError()